src/core/manager.rs

```diff
@@ -735,6 +765,59 @@ impl<'a> WindowManager<'a> {
         }
     }
 
+    /// Get a vector of references to the Clients found matching 'selector'.
+    /// The resulting vector is sorted by Client id.
+    pub fn all_clients(&self, selector: &Selector<Client>) -> Vec<&Client> {
+        match selector {
+            Selector::Focused => self.focused_client().into_iter().collect(),
+            Selector::WinId(id) => self.client_map.get(&id).into_iter().collect(),
+            Selector::Condition(f) => {
+                let mut clients = self
+                    .client_map
+                    .iter()
+                    .flat_map(|(_, v)| if f(v) { Some(v) } else { None })
+                    .collect::<Vec<_>>();
+                clients.sort_by(|a, b| a.id().cmp(&b.id()));
```

Comment:
I chose to use `sort_by` instead of implementing `PartialOrd`, `Eq` and `Ord` on `Client`. Alternatively it might make sense to use `BTreeMap` instead of `HashMap`, because B Trees are ordered. Its `Iterator` is sorted by the key, which would be `WinId`. The downside would be that it has a slightly worse time complexity, but benchmarks show that the difference isn't that much worse in practice.

Reply:
Ah that's a good point, yeah a BTree might be the way to go...operations on the map are not really going to be the main bottleneck given all the interaction with the X server. I'll take a look!